File: backend/app/schemas/school_teacher.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class InviteTeacherRequest(BaseModel):
    teacher_email: str


class SchoolTeacherResponse(BaseModel):
    id: UUID
    school_id: UUID
    teacher_id: UUID
    status: str
    invited_at: datetime
    joined_at: datetime | None


class TeacherInSchoolResponse(BaseModel):
    id: UUID  # school_teachers.id — used for suspend/remove
    email: str
    name: str
    status: str
    classes: int = 0

File: backend-java/src/main/java/com/easylearn/entity/SchoolTeacher.java
package com.easylearn.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "school_teachers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SchoolTeacher {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "school_id", nullable = false)
    private UUID schoolId;

    @Column(name = "teacher_id", nullable = false)
    private UUID teacherId;

    @Column(nullable = false)
    @Builder.Default
    private TeacherStatus status = TeacherStatus.INVITED;

    @Column(name = "invited_at", updatable = false)
    private LocalDateTime invitedAt;

    @Column(name = "joined_at")
    private LocalDateTime joinedAt;

    @PrePersist
    private void prePersist() {
        if (invitedAt == null) invitedAt = LocalDateTime.now();
    }
}
